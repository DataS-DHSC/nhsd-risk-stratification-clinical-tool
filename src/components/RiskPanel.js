import React from 'react';
import PropTypes from 'prop-types';

const RiskPanel = ({ riskType, probability, relativeRisk }) => (
  <>
    <ul>
      <li id={`absolute-risk-${riskType}`}>
        Absolute risk:{' '}
        <strong>{`${probability}% (1 in ${(100 / probability).toFixed(
          0
        )})`}</strong>
      </li>
      <li id={`relative-risk-${riskType}`}>
        Relative risk: <strong>{relativeRisk}</strong>
      </li>
    </ul>
  </>
);

RiskPanel.propTypes = {
  riskType: PropTypes.string.isRequired,
  probability: PropTypes.number.isRequired,
  relativeRisk: PropTypes.string.isRequired,
};

export default RiskPanel;
