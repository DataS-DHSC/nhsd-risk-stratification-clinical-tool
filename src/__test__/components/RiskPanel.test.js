import React from 'react';
import { render } from '@testing-library/react';
import RiskPanel from '../../components/RiskPanel';

const setup = () => {
  const props = {
    riskType: 'hospitalisation',
    probability: 5,
    relativeRisk: 50,
  };

  return render(<RiskPanel {...props} />);
};

describe('RiskPanel', () => {
  it('should match snapshot ', () => {
    const container = setup();
    expect(container.asFragment()).toMatchSnapshot();
  });
});
