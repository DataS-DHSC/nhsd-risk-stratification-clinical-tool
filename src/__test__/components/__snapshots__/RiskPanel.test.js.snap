// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiskPanel should match snapshot  1`] = `
<DocumentFragment>
  <ul>
    <li
      id="absolute-risk-hospitalisation"
    >
      Absolute risk: 
      <strong>
        5% (1 in 20)
      </strong>
    </li>
    <li
      id="relative-risk-hospitalisation"
    >
      Relative risk: 
      <strong>
        50
      </strong>
    </li>
  </ul>
</DocumentFragment>
`;
